### PR TITLE
Bz 2153469 failover logic

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
@GowthamShanmugam , @SanjalKatiyar Please review

File: packages/mco/components/modals/app-failover-relocate/error-messages.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    ),
+    variant: AlertVariant.danger,
+  },
+  [ErrorMessageType.TARGET_CLUSTER_IS_NOT_AVAILABLE]: {
+    title: t('Target cluster is offline.'),
+    message: (
+      <Trans t={t}>
+        <p>
+          To begin failover, the target cluster must be available. To bring the
+          cluster online, refer to the instructions in the documentation
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/app-failover-relocate/error-messages.tsx:127]:
```suggestion
        <p>
            To begin failover, the target cluster must be available. Check the
            status and try again. If the managed cluster status is offline, follow
            the instructions in the documentation
```

File: packages/mco/components/modals/app-failover-relocate/error-messages.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        <p>
+          The status for both the primary and target clusters must be available
+          for relocating. To bring the cluster online, refer to the instructions
+          in the documentation
+          <ViewDocumentation
+            doclink={DOC_LINKS.ACM_OFFLINE_CLUSTER}
+            text="Troubleshoot"
+          />
+        </p>
+      </Trans>
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/app-failover-relocate/error-messages.tsx:117]:
```suggestion
      <Trans t={t}>
        <ul>
          <li>
            The status for both the primary and target clusters must be
            available for relocating. Check the status and try again.
          </li>
          <li>
            To bring the cluster online, refer to the instructions in the
            documentation
            <ViewDocumentation
              doclink={DOC_LINKS.ACM_OFFLINE_CLUSTER}
              text="Troubleshoot"
            />
          </li>
        </ul>
      </Trans>
```

File: packages/mco/utils/doc-utils.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { Text, TextVariants } from '@patternfly/react-core';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+
+export const ODF_DOC_VERSION = '4.13';
```
Comment (GowthamShanmugam) [inline packages/mco/utils/doc-utils.tsx:6]: TODO: while backport this has to be changed to 4.12

File: packages/mco/utils/doc-utils.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+export const ODF_DOC_HOME = `https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/${ODF_DOC_VERSION}`;
+export const ODF_DOC_BASE_PATH = `${ODF_DOC_HOME}/html-single`;
+
+export const ACM_DOC_VERSION = '2.8';
```
Comment (GowthamShanmugam) [inline packages/mco/utils/doc-utils.tsx:10]: TODO: while backport this has to be changed to 2.7

File: packages/mco/components/modals/app-failover-relocate/target-cluster-selector.tsx [hunk 5]
```diff
@@ -215,35 +203,51 @@ export const TargetClusterSelector: React.FC<TargetClusterSelectorProps> = ({
...
-        errorMessage = !isClusterAvailable
-          ? requiredClusterStatus.errorMessage
-          : '';
+        if (state.actionType === ACTION_TYPE.RELOCATE) {
+          // Ensure origin and target both clusters are unfenced
+          const isClustersUnfenced = drClusterList.every((drCluster) =>
+            [DRClusterStatus.UNFENCED, DRClusterStatus.AVAILABLE].includes(
+              drCluster?.status?.phase as DRClusterStatus
+            )
+          );
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/app-failover-relocate/target-cluster-selector.tsx:229]:
```suggestion
          const isClustersUnfenced = drClusterList.every((drCluster) =>
            DRClusterStatus.FENCED !== drCluster?.status?.phase as DRClusterStatus
            )
          );
```

File: packages/mco/components/modals/app-failover-relocate/target-cluster-selector.tsx [hunk 6]
```diff
@@ -215,35 +203,51 @@ export const TargetClusterSelector: React.FC<TargetClusterSelectorProps> = ({
...
+          errorMessage =
+            !isClustersUnfenced && ErrorMessageType.SOME_CLUSTERS_ARE_FENCED;
+        } else {
+          // Ensure origin cluster is fenced
+          const isClustersFenced = [DRClusterStatus.FENCED].includes(
+            originCluster?.status?.phase as DRClusterStatus
+          );
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/app-failover-relocate/target-cluster-selector.tsx:236]:
```suggestion
          const isClustersFenced = DRClusterStatus.FENCED === originCluster?.status?.phase as DRClusterStatus;
```

File: packages/mco/components/modals/app-failover-relocate/target-cluster-selector.tsx [hunk 7]
```diff
@@ -215,35 +203,51 @@ export const TargetClusterSelector: React.FC<TargetClusterSelectorProps> = ({
...
+          const fencedErrorMessage =
+            !isClustersFenced && ErrorMessageType.PRIMARY_CLUSTER_IS_NOT_FENCED;
+          // Ensure target cluster is unfenced
+          const isClusterUnFenced = [
+            DRClusterStatus.UNFENCED,
+            DRClusterStatus.AVAILABLE,
+          ].includes(targetCluster?.status?.phase as DRClusterStatus);
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/app-failover-relocate/target-cluster-selector.tsx:243]:
```suggestion
          const isClusterUnFenced = DRClusterStatus.FENCED !== targetCluster?.status?.phase as DRClusterStatus;
```

**Comment by SanjalKatiyar** [comment]
/close
needs to send directly to release-4.12 branch

**Pull request closed without merging**